src/Computers.java

```diff
@@ -0,0 +1,61 @@
+public class CalculatingDevice {
+    protected String name;
+
+    CalculatingDevice(String name) {
+        this.name = name;
+    }
+    
+    public String calculate(double x, double y) {
+        double sum = x + y;
+        return name + ": " + x + "+" + y + "=" + sum;
+    }
+
+    public static void printRes(CalculatingDevice[] a, double x, double y) {
+        for (int i = 0; i < a.length; i++) {
+            System.out.println(a[i].calculate(x, y));
+        }
+    }
+}
+
+public class Calculator extends CalculatingDevice {
+    public Calculator(String name) {
+        super(name);
+    }
+
+    @Override
+    public String calculate(double x, double y) {
+        String additionResult = super.calculate(x, y);
+        double sub = x - y;
+        return additionResult + "; " + x + "-" + y + "=" + sub;
+    }
+}
+
+public class Computer extends Calculator {
+    public Computer(String name) {
+        super(name);
+    }
+
+    @Override
+    public String calculate(double x, double y) {
+        String additionAndSubtraction = super.calculate(x, y);
+        double multiplication = x * y;
+        if (y == 0) {
+            return "Division by zero error";
```

Comment:
Please improve this, so that you will persist the results of all other operations. So that you will return something like:
`Cray: 21.0+0.0=21.0; 21.0-0.0=21.0; 21.0*0.0=0.0; Cannot Divide by 0`